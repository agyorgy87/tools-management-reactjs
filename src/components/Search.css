@import url('https://fonts.googleapis.com/css2?family=Jura&display=swap');

.searchFor {
    margin-top: 20px;
    display:flex;
    color: white;
    font-family: 'Jura', sans-serif;
}

.mainSearch {
    display: flex;
}

.SearchInput {
    height: 50px;
    width: 80%;
    border: none;
    border-radius: 5px;
    outline: none;
    font-size: 30px;
    font-family: 'Jura', sans-serif;
}

.searchButton {
    width: 20%;
    margin-left: 20px;
    border: 2px solid white;
    border-radius: 10px;
    outline: none;
    background-color: #41414d;
    color: white;
    font-size: 200%;
    font-family: 'Jura', sans-serif;
}

.searchButton:hover {
    color: #fe8e22; 
}