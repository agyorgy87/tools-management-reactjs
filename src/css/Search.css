@import url('https://fonts.googleapis.com/css2?family=Jura&display=swap');

.search-for-label {
    margin-top: 20px;
    display:flex;
    color: white;
    font-family: 'Jura', sans-serif;
}

.input-button-container {
    display: flex;
}

.search-input {
    height: 50px;
    width: 80%;
    border: none;
    border-radius: 5px;
    outline: none;
    font-size: 30px;
    font-family: 'Jura', sans-serif;
}

.search-button {
    width: 20%;
    margin-left: 20px;
    border: 2px solid white;
    border-radius: 10px;
    outline: none;
    background-color: #41414d;
    color: white;
    font-size: 200%;
    font-family: 'Jura', sans-serif;
}

.search-button:hover {
    color: #fe8e22; 
}