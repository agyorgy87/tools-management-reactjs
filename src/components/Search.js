import '../css/Search.css';
import React, {useRef,useEffect,useState} from 'react';

export default function Search (props) {

    let searchBar = useRef(null);

    useEffect(() => {
        searchBar.current.focus();
    })

    let tools = props.data;
    const[allData, setAllData] = useState(tools);

    const[filterValue, setFilterValue] = useState([])

    const [inputValue, setInputValue] = useState('');

    const SearchButton = (props) => {
        setFilterValue([]);
        const TitleName = allData.filter(element => element.Name.toLowerCase().search(inputValue.toLowerCase()) !== -1);
        filterValue.push(...TitleName);
        props.setData(filterValue);
        
    }   

    return (  
        <div>
            <label htmlFor="searchit" className="search-for-label">search for:</label>
                <div className="input-button-container">
                    <input 
                    placeholder='for example "hammer" '
                    className="search-input" 
                    id="searchit" 
                    name="searchit" 
                    ref={searchBar}
                    onChange={event => setInputValue(event.target.value)}
                    />
                    <button 
                    className="search-button"
                    type="button"
                    onClick={() => SearchButton(props)}
                    >Search
                    </button>
                </div>
        </div>
    )
}