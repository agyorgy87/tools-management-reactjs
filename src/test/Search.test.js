import renderer from 'react-test-renderer';
import Search from "../components/Search.js";

test("test Search", () => {
    let tree = renderer
    .create(<Search data={[]} setData={() => {}}/>)
    .toJSON()
    expect(tree).toMatchSnapshot();
});
