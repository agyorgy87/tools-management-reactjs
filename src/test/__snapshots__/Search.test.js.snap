// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Search 1`] = `
<div>
  <label
    className="search-for-label"
    htmlFor="searchit"
  >
    search for:
  </label>
  <div
    className="input-button-container"
  >
    <input
      className="search-input"
      id="searchit"
      name="searchit"
      onChange={[Function]}
      placeholder="for example \\"hammer\\" "
    />
    <button
      className="search-button"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
